src/app/api/v1/user/welcome/route.ts

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getToken } from "next-auth/jwt";
+import { NextRequest, NextResponse } from "next/server";
+import AppConstants from "../../../../../appConstants";
+import { getL10n } from "../../../../functions/server/l10n";
+import {
+  getSubscriberByEmail,
+  setOnerepProfileId,
+} from "../../../../../db/tables/subscribers";
+
+async function callOneRep(method: string, path: string, body: string) {
+  const bearerToken = process.env.ONEREP_API_KEY;
+  const options = {
+    method,
+    headers: {
+      Authorization: `Bearer ${bearerToken}`,
+      "Content-Type": "application/json",
+    },
+  };
+  if (method !== "GET" && method !== "HEAD") {
+    //@ts-ignore FIXME
+    options.body = body;
+  }
+  const result = await fetch(`${process.env.ONEREP_API_BASE}/${path}`, options);
+  if (!result.ok) {
+    throw new Error("Error connecting to provider");
+  }
+  return result.json();
+}
```

Comment:
This looks very similar to (but not exactly the same as) `callOneRep()` method https://github.com/mozilla/blurts-server/blob/911c5ec128440278cd90e66f9d37e719374080f8/src/app/(proper_react)/redesign/(authenticated)/user/welcome/scanning/page.tsx#L44-L65
I think the only difference is one function handles an optional "POST" `body` payload and the other doesn't.
Q: Should we make this some shared utility lib function somewhere and share between the two files?

Also, should `method` be a generic `:string` type, or should we hardcode constant `"GET"|"POST"`? Those were the only two methods I found in use (via <kbd>git grep -n -A1 "callOneRep(" src</kbd>) although the code above does a check for `!== "HEAD"`).

Reply:
I forgot that @Vinnl already landed a very similar set of functions so I switched to that, and his is much nicer :)